lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -71,13 +80,53 @@ define((require, exports, module) ->
             left: leftNearest
             right: rightNearest
             mismatch: true
-        if result.left and result.right
-            expectedRightBracket = session.$brackets[session.getLine(leftNearest.row).charAt(leftNearest.column)]
-            rightBracket = session.getLine(rightNearest.row).charAt(rightNearest.column)
+            equals: (object) ->
+
+                if object.left != @left
+                    return false
+                if object.right != @right
+                    return false
+                if object.mismatch != @mismatch
+                    return false
+                return true
+        
+        if result.left && result.right
+            expectedRightBracket = session.$brackets[session.getLine(result.left.row).charAt(result.left.column)]
+            rightBracket = session.getLine(result.right.row).charAt(result.right.column)
             if  expectedRightBracket == rightBracket
                 result.mismatch = false
         return result
 
-    exports.highlighter = 
+    init = (editor, bindKey) ->
+        session = editor.getSession()
+        keyboardHandler = 
+            name: 'highlightBrackets'
+            bindKey: bindKey
+            exec: (editor)  -> return highlightBrackets(editor)
+            readOnly: true
+
+
+        editor.commands.addCommand(keyboardHandler);
+
+        session.getSelection().on("changeCursor", -> 
+            if session.$bracketHighlightLeft || session.$bracketHighlightRight 
+                session.removeMarker(session.$bracketHighlightLeft)
+                session.removeMarker(session.$bracketHighlightRight)
+                session.$bracketHighlightLeft = null
+                session.$bracketHighlightRight = null
+                if (!isInsideCurrentHighlight())
+                    highlightBrackets(editor)
+            return
+        )
+
+        isInsideCurrentHighlight = -> 
+            oldPosition = session.$highlightRange;
```

Comment:
I am being annoying, but anyway, don't you feel a little bit uncomfortable looking at the code where position is assigned a range?